### DISCUSSION TITLE
Is this filtering possible?

### DISCUSSION THREAD

**Discussion by nenadfilipovic** [prisma/prisma#9803] [Q&A] [answered]
I have requirement to filter devices by 4 categories (mine,public,private,favorite), code looks something like this:

```ts
switch (category) {
  case 'favorite':
    if (viewer) {
      result.push({
        OR: {
          deviceOnFavoriteDevice: {
            some: { userId: viewer.id },
          },
        },
      });
    }
    break;
  case 'public':
    result.push({ OR: { isPublic: { equals: true } } });
    break;
  case 'private':
    result.push({ OR: { isPublic: { equals: false } } });
    break;
  case 'mine':
    if (viewer) {
      result.push({ OR: { userId: { equals: viewer.id } } });
    }
    break;
}
```

Ofcourse it does not work, what I need to achieve is, if categories array contains `mine` it should only get devices with my id, if it contains `mine and favorite` it should get mine and my favorite devices, if it contains `private, public and mine` it should get all private, mine and public without duplicates. I need somehow to discriminate rest of rows after each category is checked. I probably should go first with checking `mine` then `favorite` after that `public` and at the end `private`. What would be good way to achieve this without multiple queries? Thanks.

**Comment by nenadfilipovic** [comment, marked as answer]
Something like this seems to work:

```ts
const isPublic = { isPublic: { equals: true } };
const isPrivate = { isPublic: { equals: false } };
const isFavorite = {
  deviceOnFavoriteDevice: {
    some: { userId: viewer.id },
  },
};
const isMine = { userId: { equals: viewer.id } };
```

```ts
switch (category) {
  case 'mine':
    result.push(isMine);
    break;
  case 'favorite':
    result.push(isFavorite);
    break;
  case 'public':
    result.push({
      AND: [
        { ...isPublic },
        { NOT: { ...isMine } },
        { NOT: { ...isFavorite } },
      ],
    });
    break;
  case 'private':
    result.push({
      AND: [
        { ...isPrivate },
        { NOT: { ...isMine } },
        { NOT: { ...isFavorite } },
      ],
    });
    break;
  default:
    break;
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
